payment/checkout.py

```diff
@@ -1,70 +1,159 @@
+import uuid
 from flask import (
+    Markup,
     Blueprint,
     render_template,
     request,
-    session,
     current_app,
     flash,
     redirect,
     url_for,
 )
-from itertools import accumulate
 import omise
 from store.cart import Cart, Price
-import uuid
 
 checkout = Blueprint("checkout", __name__, template_folder="templates")
 
 
+def get_client_ip():
+    if "X-Forwarded-For" in request.headers:
+        return (
+            next(iter(request.headers["X-Forwarded-For"].split(",")))
+            or request.remote_addr
+        )
+    else:
+        return request.remote_addr
+
+
+def processed(chrg, already_redirected=False):
+    """
+    Process charge depending on payment method.  `already_redirected`
+    tries to ensure that the customer is redirected only once.
+    """
+
+    cart = Cart()
+    order_id = chrg._attributes["metadata"]["order_id"]
```

Comment:
Is there a better way to access these attributes?

Reply:
@danfowler , Do you have any concern about this way ?